### PR TITLE
http2: Add compliance test for header values

### PR DESCRIPTION
**Pull request by yanavlasov**
Risk Level: Low, Test Only
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: envoy/http/header_map.h [hunk 1]
```diff
@@ -231,6 +236,11 @@ class HeaderString {
 
   bool valid() const;
 
+  // Test only method that does not have validation and allows setting arbitrary values.
+  // Test code that needs access to it is declared below.
+  friend class Http2::Http2CodecImplTest;
+  void setCopyUnvalidated(absl::string_view view);
```
Comment (zuercher) [inline envoy/http/header_map.h:242]: Should we name this `setCopyUnvalidateForTest`? I think we've done this in other places to re-enforce the test-only status of the method.

Comment (yanavlasov) [reply envoy/http/header_map.h]: Done

**Review by zuercher — APPROVED** [review]
Thanks!

**Pull request merged**